FWCore/ParameterSet/python/SequenceTypes.py

```diff
@@ -187,19 +189,36 @@ class _ModuleSequenceType(_ConfigureComponent, _Labelable):
     def __init__(self,*arg, **argv):
         self.__dict__["_isFrozen"] = False
         self._seq = None
-        if len(arg) > 1:
+        if (len(arg) > 1 and not isinstance(arg[1], Task)) or (len(arg) > 0 and not isinstance(arg[0],_Sequenceable) and not isinstance(arg[0],Task)):
             typename = format_typename(self)
             msg = format_outerframe(2) 
-            msg += "%s takes exactly one input value. But the following ones are given:\n" %typename
+            msg += "The %s constructor takes zero or one sequenceable argument followed by zero or more arguments of type Task. But the following types are given:\n" %typename
             for item,i in zip(arg, xrange(1,20)):
-                msg += "    %i) %s \n"  %(i, item._errorstr())
-            msg += "Maybe you forgot to combine them via '*' or '+'."     
+                try:
+                    msg += "    %i) %s \n"  %(i, item._errorstr())
+                except:
+                    msg += "    %i) Not sequenceable and not a Task\n" %(i)
+            if len(arg) > 1 and isinstance(arg[0],_Sequenceable) and isinstance(arg[1], _Sequenceable):
+                msg += "Maybe you forgot to combine the sequenceable arguments via '*' or '+'."
             raise TypeError(msg)
-        if len(arg)==1:
-            _checkIfSequenceable(self, arg[0])
+        tasks = arg
+        if len(arg) > 0 and isinstance(arg[0], _Sequenceable):
             self._seq = _SequenceCollection()
             arg[0]._appendToCollection(self._seq._collection)
+            tasks = arg[1:]
         self._isModified = False
+        self._tasks = set()
+        self._orderedTasks = list()
```

Comment:
In the PR as currently written, the elements of a Task, and the tasks in Paths, EndPaths, Sequences and Schedules are held in a set to prevent duplicate entries and also held in a list that is used for almost everything else. The main purpose of also having a list is to keep the iteration order fixed and reproducible. So for example if a user runs a visitor over a Path, that user would see the Tasks and contents of the Task always in the same order. That order would not depend on the order of iteration over a Python set which is not fixed and could vary. In a lot of cases this variation in order would not matter. In principle cmsRun will not be affected in any way by the order of Tasks or the order of the items on a Task. But that is not actually true when one considers the complicated ways the various CMS defined python functions that replace modules and modify configurations work. One can imagine pathological cases where the order does matter when various replace and delete and expansion operations are going on. And even something simple like printing out a text file from a Python visitor and trying to use it as a reference and "diff" against it might fail if the visitation order depended on the iteration order of Python sets.